Sources/Overture/Optional.swift

```diff
@@ -22,3 +22,23 @@ public func map<A, B>(
 
     return { try $0.map(transform) }
 }
+
+
+/// Transforms a pair of optionals into an optional pair.
+///
+/// - Parameters:
+///   - a: An optional value.
+///   - b: Another optional value.
+/// - Returns: An optional pair of values.
+public func zip<A, B>(_ a: A?, _ b: B?) -> (A, B)? {
```

Comment:
any reason to not throw this in `ZipOptional.swift`?

Reply:
Just localizing all the higher-order zips in their own file. That way we can just delete them if/when we get variadic generics.